### ISSUE TITLE
Only Printing notations break parsing by creating keywords

### ISSUE THREAD

**Issue by andres-erbsen** [coq/coq#9741]
#### Description of the problem

```coq
Notation "'a'" := tt (only printing).
Goal True. let a := constr:(1+1) in idtac a. Abort.
```

#### Coq Version

8.9.0

**Comment by samuelgruetter** [comment]
Just ran into this again, here's another test case:

```coq
Require Import Coq.Strings.String.
Require Import Coq.ZArith.ZArith.
Open Scope string_scope.

Axiom Ox: string -> Z.

Axiom isMMIOAddr: Z -> Prop.

Notation "'Ox' a" := (Ox a) (only printing, at level 10, format "'Ox' a").

Goal False.
  set (f := isMMIOAddr).
  set (x := f (Ox "0018")).
```

says

```
Error: Syntax error: [constr:operconstr level 200] expected after '(' (in [constr:operconstr]).
```

It works if I remove the `Notation`, though.

This is very confusing: `only printing` should really mean "does not affect parsing".

**Comment by samuelgruetter** [comment]
Do you think there would be an easy fix for this @herbelin?

**Comment by herbelin** [comment]
You are lucky, this happens to be apparently fixed by #10832. I just updated #10832 to reflect this.
